src/metatensor/models/soap_bpnn/model.py

```diff
@@ -93,11 +94,25 @@ def forward(self, features: TensorMap) -> TensorMap:
 
 class Model(torch.nn.Module):
     def __init__(
-        self, all_species: List[int], hypers: Dict = DEFAULT_MODEL_HYPERS
+        self, capabilities: ModelCapabilities, hypers: Dict = DEFAULT_MODEL_HYPERS
```

Comment:
I don't think we should have a docstring for the SOAP-BPNN model specifically, because all models will have the same __init__ and forward. However, if you want, we can have a fake model class somewhere that documents the behavior of these two functions for any model